keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/LambdaModelVertex.java

```diff
@@ -0,0 +1,93 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A vertex whose operation is the execution of a lambda.
+ *
+ * It is able to execute a lambda and is able to parse the result.
+ *
+ * It stores multiple output values and a model result vertex is required to extract a specific value.
+ */
+public class LambdaModelVertex extends DoubleVertex implements ModelVertex<DoubleTensor> {
+
+    private static final DoubleTensor MODEL_RETURN_VALUE = DoubleTensor.scalar(0.);
+
+    private Map<VertexLabel, Vertex<? extends Tensor>> inputs;
+    private Map<VertexLabel, Tensor> outputs;
+    private Consumer<Map<VertexLabel, Vertex<? extends Tensor>>> executor;
+    private Function<Map<VertexLabel, Vertex<? extends Tensor>>, Map<VertexLabel, Tensor>> extractOutput;
+    private boolean hasCalculated;
+
+    public LambdaModelVertex(Map<VertexLabel, Vertex<? extends Tensor>> inputs,
+                             Consumer<Map<VertexLabel, Vertex<? extends Tensor>>> executor,
+                             Function<Map<VertexLabel, Vertex<? extends Tensor>>, Map<VertexLabel, Tensor>> extractOutput) {
```

Comment:
What's the thinking behind splitting up the executor and the extractOutput functions separate?

Reply:
To make what's going on as clear to the user as possible. By splitting it up it makes it clear that they need to run some command and then extract the outputs somehow. Whereas if we required one argument that had to execute and get outputs it would be a bit of a daunting task.